ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/IntegrationDetailPage.java

```diff
@@ -20,11 +21,13 @@
 
 	private static final class Element {
 		public static final By ROOT = By.cssSelector("syndesis-integration-detail-page");
-
 		public static final By STATUS = By.cssSelector("syndesis-integration-status");
 		public static final By TITLE = By.cssSelector("h1");
 	}
 
+	@Getter
+	private IntegrationDetailPageFactory detailPageFactory = new IntegrationDetailPageFactory();
```

Comment:
I dont think that having factory as instance variable does make much sense. Maybe making special getter for getting step by type would be better.

Reply:
@lvydra to align it with detail/edit/list pages in connections/integrations

Reply:
@Stefan365  But factory class is not a page object, only delivers you instantion of page object depending on its type. So, you probably should put there step page object class and instantiate it in get method trough factory.

Reply:
@lvydra sorry - I placed above comment to wrong place. (above comment should have relate to renaming ConnectionCreatePage to ConnectionEditPage)

Reply:
@lvydra to the private factory:
this should express that products of this factory are components of EditPage (general philosophy applied here in thie refactor). It could be done also as you propose.

Reply:
I understand you, but factory itself is only a tool to get them. If you put there general ancestor instead, it will probably represent that idea better.

Reply:
👍